src/modules/dashboard/components/ColonyHome/ColonyFunding/ColonyFunding.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useCallback, useMemo } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+
+import Button from '~core/Button';
+import { DialogType } from '~core/Dialog';
+import withDialog from '~core/Dialog/withDialog';
+import Heading from '~core/Heading';
+import { Address } from '~types/index';
+import { useSelector, useRoles } from '~utils/hooks';
+
+import TokenItem from './TokenItem';
+import { useColonyTokens } from '../../../hooks/useColonyTokens';
+import { canMoveTokens as canMoveTokensCheck } from '../../../../admin/checks';
+import { currentUserSelector } from '../../../../users/selectors';
+
+import styles from './ColonyFunding.css';
+
+const MSG = defineMessages({
+  buttonFund: {
+    id: 'dashboard.ColonyHome.ColonyFunding.buttonFund',
+    defaultMessage: 'Fund',
+  },
+  title: {
+    id: 'dashboard.ColonyHome.ColonyFunding.title',
+    defaultMessage: 'Available Funding',
+  },
+});
+
+interface Props {
+  colonyAddress: Address;
+  currentDomainId: number;
+  openDialog: (dialogName: string, dialogProps?: object) => DialogType;
+}
+
+const displayName = 'dashboard.ColonyHome.ColonyFunding';
+
+const ColonyFunding = ({
+  colonyAddress,
+  currentDomainId,
+  openDialog,
+}: Props) => {
+  const [tokenReferences, tokens] = useColonyTokens(colonyAddress);
+
+  const {
+    profile: { walletAddress: currentUserWalletAddress },
+  } = useSelector(currentUserSelector);
+  const { data: roles } = useRoles(colonyAddress);
+  const canMoveTokens = useMemo(
+    () => canMoveTokensCheck(roles, currentUserWalletAddress),
+    [currentUserWalletAddress, roles],
+  );
+
+  const handleMoveTokens = useCallback(
+    () =>
+      openDialog('TokensMoveDialog', {
+        colonyAddress,
+        toDomain: currentDomainId,
+      }),
+    [openDialog, colonyAddress, currentDomainId],
+  );
+
+  // hide for root domain
+  return currentDomainId === 0 ? null : (
```

Comment:
Hm yeah this is actually kind of confusing behaviour now that I think about it more, but it's what was agreed with product at least for now. I think the whole selecting of domain + hierarchy + "all domains" behaviour needs another look from dev + product point of view.